### DISCUSSION TITLE
Can i set defaultSort to relationship column?

### DISCUSSION THREAD

**Discussion by hazwan91** [filamentphp/filament#8484] [Help] [answered]
How to set defaultSort using `relationship column` ?

I am currently creating a user list and i don't want to sort by user name by default. I want to sort by `role name` which is from `Role` model

Normally i would set it like `->defaultSort('name')` but i can't use such as this `->defaultSort('role.name')`
I also try to use closure inside the defaultSort such as:

```
->defaultSort(function() {
    return 'name'
))
```
but no changes.

I can only find defaultSort method in 1 place on the docs.

I don't want to use sortable since i want to make it as default as soon as the table loaded.

Any ideas?

**Comment by leandrocfe** [comment, marked as answer]
You can use [customizing-the-resource-eloquent-query](https://filamentphp.com/docs/3.x/panels/resources/getting-started#customizing-the-resource-eloquent-query)

```php
public static function getEloquentQuery(): Builder
{
    return parent::getEloquentQuery()
        ->withAggregate('roles', 'name')
        ->orderBy('roles_name', 'asc');
}
```

**Comment by hazwan91** [comment]
Thanks @leandrocfe for pointing me to right direction. I am still learning the filament way and not dare to customize a lot.
